src/scripts/main.js

```diff
@@ -1,11 +1,25 @@
 'use strict';
 
 function waitFor(element, eventName) {
-  // write your code here
+  if (element) {
+    return new Promise((resolve, reject) => {
+      element.addEventListener(eventName, () => {
+        const string = `It was ${eventName} on the element: ${element.nodeName}, id: ${element.id}.`;
+
+        console.log(eventName, element.id);
+        resolve(string);
+      });
+    });
+  }
 }
 
 const printMessage = (message) => {
-  // write your code here
+  const form = document.querySelector('form');
```

Comment:
This code selects the form each time the function `printMessage` is called. To improve performance, consider selecting the form once and storing it in a variable outside the function.